decryptor/postgresql/data_encoder.go

```diff
@@ -0,0 +1,255 @@
+package postgresql
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"errors"
+	"github.com/cossacklabs/acra/decryptor/base"
+	"github.com/cossacklabs/acra/encryptor"
+	"github.com/cossacklabs/acra/encryptor/config"
+	common2 "github.com/cossacklabs/acra/encryptor/config/common"
+	"github.com/cossacklabs/acra/logging"
+	"github.com/cossacklabs/acra/utils"
+	"github.com/sirupsen/logrus"
+	"strconv"
+)
+
+// ErrInvalidDataEncoderMode unsupported DataEncoderMode value
+var ErrInvalidDataEncoderMode = errors.New("unsupported DataEncoderMode value")
+
+// ErrInvalidIntValueBinarySize unsupported DataEncoderMode value
+var ErrInvalidIntValueBinarySize = errors.New("unsupported binary size of int value")
+
+// DataEncoderMode mode of PgSQLDataEncoderProcessor
+type DataEncoderMode int8
+
+// Available modes of DataEncoderMode
+const (
+	DataEncoderModeEncode = iota
+	DataEncoderModeDecode
+)
+
+// PgSQLDataEncoderProcessor implements processor and encode/decode binary intX values to text format which acceptable by Tokenizer
+type PgSQLDataEncoderProcessor struct {
+	mode DataEncoderMode
+}
+
+// NewPgSQLDataEncoderProcessor return new data encoder/decoder from/to binary format for tokenization
+func NewPgSQLDataEncoderProcessor(mode DataEncoderMode) (*PgSQLDataEncoderProcessor, error) {
+	switch mode {
+	case DataEncoderModeDecode, DataEncoderModeEncode:
+		return &PgSQLDataEncoderProcessor{mode}, nil
+	}
+	return nil, ErrInvalidDataEncoderMode
+}
+
+// ID return name of processor
+func (p *PgSQLDataEncoderProcessor) ID() string {
+	return "PgSQLDataEncoderProcessor"
+}
+
+// here we process encryption/tokenization results before send it to a client
+// acra decrypts or de-tokenize SQL literals, so we should convert string SQL literals to binary format
+// if client expects int, then parse INT literals and convert to binary 4/8 byte format
+// if expects bytes, then pass as is
+// if expects string, then leave as is if it is valid string or encode to hex
+// if it is encrypted data then we return default values or as is if applicable (binary data)
+func (p *PgSQLDataEncoderProcessor) encodeBinary(ctx context.Context, data []byte, setting config.ColumnEncryptionSetting, columnInfo base.ColumnInfo, logger *logrus.Entry) (context.Context, []byte, error) {
```

Comment:
Seems to be `columnInfo` is unused field here.

Reply:
yes. but contains metadata about columns and left because may be left for future use. It wasn't added in this PR, it already exists. I suggest to review need of this struct at finish of this feature. And if it will be redundant then remove.